pyqtorch/noise.py

```diff
@@ -0,0 +1,420 @@
+
+    .. math::
+        \\rho \\Rightarrow (1-p) \\rho + p X \\rho X^{\\dagger}
+
+    Args:
+        target (int): The index of the qubit being affected by the noise.
+        probability (float): The probability of a bit flip error.
+
+    Raises:
+        TypeError: If the probability value is not a float.
+    """
+
+    def __init__(
+        self,
+        target: int,
+        probability: float,
+    ):
+        # Verification probability type:
+        if not isinstance(probability, float):
+            raise TypeError("The probability values must be float")
+
+        # Define Kraus operator:
+        K0: Tensor = torch.sqrt(torch.tensor(1 - probability, dtype=DEFAULT_MATRIX_DTYPE)) * IMAT
+        K1: Tensor = torch.sqrt(torch.tensor(probability, dtype=DEFAULT_MATRIX_DTYPE)) * XMAT
+        Kraus_Bitflip: List[Tensor] = [K0, K1]
+        super().__init__(Kraus_Bitflip, target, probability)
+
+
+class PhaseFlip(Noise):
+    """
+    Initialize the PhaseFlip gate
+
+    The phase flip channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow (1-p) \\rho + p Z \\rho Z^{\\dagger}
+
+    Args:
+        target (int): The index of the qubit being affected by the noise.
+        probability (float): The probability of phase flip error.
+
+    Raises:
+        TypeError: If the probability value is not a float.
+    """
+
+    def __init__(
+        self,
+        target: int,
+        probability: float,
+    ):
+        # Verification probability type:
+        if not isinstance(probability, float):
+            raise TypeError("The probability values must be float")
+
+        # Define Kraus operator:
+        K0: Tensor = torch.sqrt(torch.tensor(1 - probability, dtype=DEFAULT_MATRIX_DTYPE)) * IMAT
+        K1: Tensor = torch.sqrt(torch.tensor(probability, dtype=DEFAULT_MATRIX_DTYPE)) * ZMAT
+        Kraus_Phase: List[Tensor] = [K0, K1]
+        super().__init__(Kraus_Phase, target, probability)
+
+
+class Depolarizing(Noise):
+    """
+    Initialize the Depolarizing gate.
+
+    The depolarizing channel is defined as:
+
+    .. math::
+        \\rho \\Rightarrow (1-p) \\rho
+            + p/3 X \\rho X^{\\dagger}
+            + p/3 Y \\rho Y^{\\dagger}
+            + p/3 Z \\rho Z^{\\dagger}
+
+    Args:
+        target (int): The index of the qubit being affected by the noise.
+        probability (float): The probability of depolarizing error.
+
+    Raises:
+        TypeError: If the probability value is not a float.
+    """
+
+    def __init__(
+        self,
+        target: int,
+        probability: float,
+    ):
+        # Verification probability type:
+        if not isinstance(probability, float):
+            raise TypeError("The probability values must be float")
+
+        # Define Kraus operator:
+        K0: Tensor = torch.sqrt(torch.tensor(1 - probability, dtype=DEFAULT_MATRIX_DTYPE)) * IMAT
+        K1: Tensor = torch.sqrt(torch.tensor(probability / 3, dtype=DEFAULT_MATRIX_DTYPE)) * XMAT
```

Comment:
I would create a tensor object once and reuse it for these tensor probabilities.

Reply:
Done.